File: go/002.go
package main

import (
    "fmt"
)

func fib (num int) int {
    var slot1 int = 1
    var slot2 int = 1
    var placeholder int
    var sum int = 0;
    for slot2 < num {
        placeholder = slot1 + slot2
        slot1 = slot2
        slot2 = placeholder
        if (slot2 % 2 == 0) {
            sum += slot2
        }
    }
    return sum
}


func main () {
    fmt.Println(fib(4000000))
}

File: js/002.js
var slot1 = 1;
var slot2 = 2;
var sum = 2;

while (slot2 < 4000000) {
    var temp = slot1 + slot2;
    slot1 = slot2;
    slot2 = temp;
    if (slot2 % 2 == 0) {
        sum += slot2;
    }
}
console.log(sum);
